engine/src/main/java/org/terasology/engine/world/block/BlockArea.java

```diff
@@ -354,4 +355,35 @@ public int hashCode() {
     public String toString() {
         return "BlockArea[(" + this.minX + ", " + this.minY + ")...(" + this.maxX + ", " + this.maxY + ")]";
     }
+
+    private int calDis(int x, int y){
+        return (int) Math.sqrt(x * x + y * y);
+    }
+    public int distanceSquared(int px, int py){
+        if(px < minX){
+            if(py < minY) {
+                return calDis(minX - px, minY - py);
+            }
+            if(py <= maxY) {
+                return minX - px;
+            }
+            return calDis(minX - px, maxY - py);
+        }else if(px <= maxX){
+            if(py < minY) {
+                return minY - py;
+            }
+            if(py <= maxY) {
+                return 0;
+            }
+            return py - maxY;
+        }else{
+            if(py < minY) {
+                return calDis(maxX - px, minY - py);
+            }
+            if(py <= maxY) {
+                return px - maxX;
+            }
+            return calDis(maxX - px, maxY - py);
+        }
+    }
```

Comment:
do we want to use block locations or the closes edge of the area. @skaldarnar 

I'm assuming this is more useful for the actual bounds of the blockarea not just the location of the blocks. 

block area is defined as (minx - .5f ,miny - .5f ) and (minx + .5f ,miny + .5f )

Reply:
Good questions. I guess that we want to return a distance of 0 in case the given point is "touching" the area.

```java
new BlockArea(0, 0, 2, 2).distanceSquared(2, 3) == 0;
new BlockArea(0, 0, 2, 2).distanceSquared(3, 2) == 0;
```

What about if the location "touches" only at the corner? 

```java
new BlockArea(0, 0, 2, 2).distanceSquared(3, 3) == ???;
```